src/main/java/seedu/address/ui/MainWindow.java

```diff
@@ -176,6 +176,15 @@ private CommandResult executeCommand(String commandText) throws CommandException
                 handleSwitchView(commandResult.getTargetView().trim());
             }
 
+            if (commandResult.isUndo()) {
```

Comment:
State should be in model, not in logic, then you wouldnt need to change the CommandResult file because you should be able to do it in the execute command of the command file which takes in model.